podman/client.py

```diff
@@ -62,15 +62,21 @@ def __init__(self, **kwargs) -> None:
 
         api_kwargs = kwargs.copy()
 
-        if "connection" in api_kwargs:
-            connection = config.services[api_kwargs.get("connection")]
-            api_kwargs["base_url"] = connection.url.geturl()
-
-            # Override configured identity, if provided in arguments
-            api_kwargs["identity"] = kwargs.get("identity", str(connection.identity))
-        elif "base_url" not in api_kwargs:
-            path = str(Path(get_runtime_dir()) / "podman" / "podman.sock")
-            api_kwargs["base_url"] = "http+unix://" + path
+        if not api_kwargs:
```

Comment:
I think add a check here to ensure that `config` is also not empty and see if that helps fix the failing tests.